.github/workflows/default.yml

```diff
@@ -342,3 +343,47 @@ jobs:
           LINKS_MYSQL_PASSWORD: links
           LINKS_MYSQL_PORT: ${{ job.services.mysql.ports[3306] }}
         shell: bash
+
+  end-to-end:
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - ubuntu-20.04
+        ocaml-compiler:
+          - 4.08.0
+        browser:
+          - firefox
+          - chrome
+    runs-on: ${{ matrix.os }}
+    env:
+      BROWSER: ${{ matrix.browser }} 
+      NPM_DIR: tests/end-to-end
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Use OCaml ${{ matrix.ocaml-compiler }}
+        uses: ocaml/setup-ocaml@v2
+        with:
+          ocaml-compiler: ${{ matrix.ocaml-compiler }}
+
+      - run: opam install -y ounit2
+
+      - name: Install Links dependencies
+        run: >-
+          opam install
+          ./links.opam
+          --deps-only
+
+      - name: Build Links from source
+        run: |
+          eval $(opam env)
+          make all-ci
+        shell: bash
+
+      - uses: actions/setup-node@v2
+      - run: npm install
+        working-directory: ${{ env.NPM_DIR }}
+      - run: npm test
+        working-directory: ${{ env.NPM_DIR }}
```

Comment:
Remove the submodule "links-database-setup" from revision control. It was added in commit 04337e.

Reply:
I can't seem to find any mention of `links-database-setup` nor the commit hash. Please double-check

Reply:
It is in this patch. Look in "Files Changed" tab, there you will see it.